homeassistant/components/mqtt/models.py

```diff
@@ -289,12 +295,51 @@ def async_render_with_possible_json_value(
         return rendered_payload
 
 
+class EntityMonitor:
+    """Monitors entity state changes."""
+
+    __slots__ = ("_entity", "_attributes")
+
+    def __init__(self, entity: Entity) -> None:
+        """Initialize entity monitor."""
+        self._entity: Entity = entity
+        self._attributes: dict[str, Any] = {}
+
+    def track(self, attributes: set[str]) -> None:
+        """Start tracking attributes."""
+        self._attributes = {
+            attribute: getattr(self._entity, attribute, UNDEFINED)
+            for attribute in attributes
+        }
+
+    @property
+    def assume_has_changed(self) -> bool:
```

Comment:
```suggestion
    def attrs_have_changed(self) -> bool:
```

Maybe `attrs_have_changed` instead since thats what this function does 99% of the time (except for the exceptional case where force_update is set)